src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                if (parts.length == 2) {
+                    String operation = parts[0].trim();
+                    int amount = Integer.parseInt(parts[1].trim());
+
+                    if (operation.equals("supply")) {
+                        supplyTotal += amount;
+                    } else if (operation.equals("buy")) {
+                        buyTotal += amount;
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error occured while reading data from: " + fromFileName, e);
+        } finally {
+            writeReport(toFileName, supplyTotal, buyTotal);
```

Comment:
this is not very correct behavior, because if we encounter an error when reading from a file, then we will send invalid data to this method and they will be written to the file